sdk/eventhub/Azure.Messaging.EventHubs/tests/Infrastructure/EventProcessorManager.cs

```diff
@@ -147,6 +147,132 @@ public Task StopAllAsync()
                 .Select(eventProcessor => eventProcessor.StopAsync()));
         }
 
+        /// <summary>
+        ///   Waits until the partition load distribution is stabilized.  Throws a <see cref="TimeoutException"/> if
+        ///   the load takes too long to stabilize.
+        /// </summary>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public async Task WaitStabilization()
+        {
+            var stabilizedStatusAchieved = false;
```

Comment:
Not critical for this PR, but I think it would be a good idea for us to talk through a bit.   There has to be a way that we can achieve stabilization in a more deterministic way than a polling loop, and I may have some ideas after doing a ton of this type of thing for the iterator's background processing.

Reply:
If you choose to stick with this for now, would you be so kind as to open an issue to track revisiting?

Reply:
Created a new issue (#7458) so we can focus on preview 3 for now. It's not that different from the old one.